### ISSUE TITLE
$in return same doc multiple time

### ISSUE THREAD

**Issue by Yimiprod** [louischatriot/nedb#315]
When a perform a request with $in i get the same doc multiple time, one for each match.

Doc
```
{
  "id": "les_4yF3O4yd",
  "someArray": [
    "sli_V1K30RIV",
    "sli_V15IiU-9",
    "sli_NyVXPo04",
    "sli_VJl3ki8V",
    "sli_4yL2Nrd4",
    "sli_EyfRd6I4",
    "sli_Nk4bSHuN",
    "sli_EkPUF-25",
    "sli_EJyCPqLN"
  ]
}
```
Request :
```
{
'someArray': {$in:  [
    "sli_V1K30RIV",
    "sli_V15IiU-9",
    "sli_NyVXPo04",
    "sli_VJl3ki8V",
    "sli_4yL2Nrd4"
  ]}
}
```
Response
```
[
  {"id": "les_4yF3O4yd"},
  {"id": "les_4yF3O4yd"},
  {"id": "les_4yF3O4yd"},
  {"id": "les_4yF3O4yd"},
  {"id": "les_4yF3O4yd"}
]
```

**Comment by louischatriot** [comment]
I can't reproduce. Pleease make sure you aare using the latest version of node and nedb, and if not solved copy paste a self containing piece of code with expected vs actual value and I'll reopen if need be.

**Comment by Yimiprod** [comment]
Here is a working example (with 1.1.2 and latest version of nedb):
```
'use strict';

var Nedb = require('nedb');
var database = new Nedb({
	filename:  'dbexample.db',
	autoload: true
});
database.ensureIndex({
	fieldName: 'structure.ref',
	unique: true
});
database.ensureIndex({
	fieldName: 'structure.meta.linkedRef'
});
var objectToStore = [
	{
		structure: {
			ref: 'les_4yF3O4yd',
			meta: {
				linkedRef: [
					'sli_4yL2Nrd4',
					'sli_EyfRd6I4',
					'sli_Nk4bSHuN',
					'sli_EkPUF-25',
					'sli_EJyCPqLN'
				]
			}
		}
	},
	{
		structure: {
			ref: 'les_2yF4O4zd',
			meta: {
				linkedRef: [
					'sli_V1K30RIV',
					'sli_V15IiU-9',
					'sli_NyVXPo04',
					'sli_VJl3ki8V',
					'sli_4yL2Nrd4'
				]
			}
		}
	}
]
database.insert(objectToStore, function(err) {
	if (!err) {
		database.persistence.compactDatafile();
	}
});
function queryData() {
	return new Promise(function(resolve, reject) {
		var request = {
			'structure.meta.linkedRef': {
				$in: [
					'sli_V1K30RIV',
					'sli_V15IiU-9',
					'sli_EkPUF-25',
					'sli_VJl3ki8V',
					'sli_4yL2Nrd4'
				]
			}
		};
		var projection = {_id: 0};
		var cursor = database.find(request, projection);

		cursor.exec(function(err, res) {
			if (err) {
				reject(err);
			}
			resolve(res, 'structure.ref');
		});
	});
}
```

**Comment by louischatriot** [comment]
This was fixed as of nedb v1.2.1. Thanks for reporting.
